js/analyzer.js

```diff
@@ -259,43 +263,83 @@ Analyzer.prototype.linkType = function(url){
+        linkFollow = "Nofollow";
     }
     return linkFollow;
 };
 
 /**
+ * checks if the url contains the keyword
+ * @param url
+ * @returns {boolean}
+ */
+Analyzer.prototype.linkKeyword = function(url){
+    var keywordFound = false;
+    var formatUrl = url.match(/<a(.*?)(?:[^>]+)?>/);
+    if(formatUrl[0].match(this.config.keyword) !== null){
+        keywordFound = true;
+    }
+    return keywordFound;
+};
+
+/**
+ * checks if the links are all followed or not, and saves this in the resultobject, to be used for scoring
+ */
+Analyzer.prototype.linkResult = function(obj){
+    var result = obj;
+    result.externalHasNofollow = false;
+    result.externalAllNofollow = false;
+    result.externalAllDofollow = false;
+    if(result.externalTotal !== result.externalDofollow && result.externalTotal > 0){
+        result.externalHasNofollow = true;
+    }
+    if(result.externalTotal === result.externalNofollow && result.externalTotal > 0){
+        result.externalAllNofollow = true;
+    }
+    if(result.externalTotal === result.externalDofollow && result.externalTotal > 0){
+        result.externalAllDofollow = true;
+    }
+    return result;
+};
+
+/**
  * counts the number of images found in a given textstring, based on the <img>-tag and returns a result object
  * @returns {{name: string, result: {total: number, alt: number, noalt: number}}}
  */
+//todo update function so it will also check on picture elements/make it configurable.
 Analyzer.prototype.imageCount = function(){
-    var imageCount = {total: 0, alt: 0, noalt: 0};
-    var imageMatches = yst_preProcessor.__store.originalText.match(/<img(?:[^>]+)?>/g);
+    var imageCount = {total: 0, alt: 0, noalt: 0, altKeyword: 0};
+    var imageMatches = this.preProcessor.__store.originalText.match(/<img(?:[^>]+)?>/g);
     if(imageMatches !== null){
         imageCount.total = imageMatches.length;
         for (var i = 0; i < imageMatches.length; i++){
-            if(this.imageAlttag(imageMatches[i])){
-                imageCount.alt++;
+            var alttag = imageMatches[i].match(/alt=([\'\"])(.*?)\1/g);
+            if(this.imageAlttag(alttag)){
+                if(this.imageAlttagKeyword(alttag)){
+                    imageCount.altKeyword++;
+                }else{
+                    imageCount.alt++;
+                }
+
             }else{
                 imageCount.noalt++;
             }
         }
     }
-    return {name: "imageCount", result: imageCount};
+    return  [ {test: "imageCount", result: imageCount } ];
 };
 
 /**
- * checks if an image has an alttag and if the alttag contains any text.
+ * checks if  the alttag contains any text.
  * @param image
  * @returns {boolean}
  */
 Analyzer.prototype.imageAlttag = function(image){
     var hasAlttag = false;
-    var alttag = image.match(/alt=['"](.*?)['"]/g);
-    if(alttag !== null){
-        if(alttag[0].split("=")[1].match(/[a-z0-9](.*?)[a-z0-9]/g) !== null){
+    if(image !== null){
```

Comment:
I would combine these two lines in one. This makes the complexity of this part lower.